datacube.py

```diff
@@ -0,0 +1,382 @@
+
+    Returns:
+    - tuple: A tuple containing the selected Sentinel-1 item and Sentinel-1 GeoDataFrame.
+    """
+
+    geom_BBOX = box(*BBOX)  # Create poly geom object from the bbox
+
+    search: pystac_client.item_search.ItemSearch = catalog.search(
+        filter_lang="cql2-json",
+        filter={
+            "op": "and",
+            "args": [
+                {
+                    "op": "s_intersects",
+                    "args": [{"property": "geometry"}, geom_BBOX.__geo_interface__],
+                },
+                {"op": "anyinteracts", "args": [{"property": "datetime"}, week]},
+                {"op": "=", "args": [{"property": "collection"}, "sentinel-1-rtc"]},
+            ],
+        },
+    )
+    s1_items = search.get_all_items()
+    print(f"Found {len(s1_items)} Sentinel-1 items")
+
+    s1_gdf = gpd.GeoDataFrame.from_features(s1_items.to_dict())
+    s1_gdf["overlap"] = s1_gdf.intersection(box(*BBOX)).area
+
+    # Choose the scene with the maximum overlap
+    selected_scene = s1_gdf.loc[s1_gdf["overlap"].idxmax()]
+    print(selected_scene)
+    selected_item_id = selected_scene["id"]
+    selected_item = catalog.get_item(selected_item_id)
+
+    return selected_item, s1_gdf
+
+
+def search_dem(BBOX, catalog, week, s2_items):
+    """
+    Search for Digital Elevation Model (DEM) items within a given bounding box (BBOX), STAC catalog, week, and Sentinel-2 items.
+
+    Parameters:
+    - BBOX (tuple): Bounding box coordinates in the format (minx, miny, maxx, maxy).
+    - catalog (pystac.Catalog): STAC catalog containing DEM items.
+    - week (str): The week in the format 'start_date/end_date'.
+    - s2_items (list): List of Sentinel-2 items.
+
+    Returns:
+    - tuple: A tuple containing DEM items and DEM GeoDataFrame.
+    """
+    search = catalog.search(
+        collections=["cop-dem-glo-30"], bbox=BBOX
+    )
+    dem_items = search.item_collection()
+    print(f"Found {len(dem_items)} items")
+
+    dem_gdf = gpd.GeoDataFrame.from_features(dem_items.to_dict())
+
+    dem_gdf.set_crs(epsg=4326, inplace=True)
+    dem_gdf = dem_gdf.to_crs(epsg=s2_items[0].properties["proj:epsg"])
+    return dem_items, dem_gdf
+
+
+def make_dataarrays(s2_items, s1_items, dem_items, BBOX, resolution, epsg):
+    """
+    Create xarray DataArrays for Sentinel-2, Sentinel-1, and DEM data.
+
+    Parameters:
+    - s2_items (list): List of Sentinel-2 items.
+    - s1_items (list): List of Sentinel-1 items.
+    - dem_items (list): List of DEM items.
+    - BBOX (tuple): Bounding box coordinates in the format (minx, miny, maxx, maxy).
+    - resolution (int): Spatial resolution.
+    - epsg (int): EPSG code for the coordinate reference system.
+
+    Returns:
+    - tuple: A tuple containing xarray DataArrays for Sentinel-2, Sentinel-1, and DEM.
+    """
+    da_sen2: xr.DataArray = stackstac.stack(
+        items=s2_items[0],
+        epsg=26910,  # UTM Zone 10N
```

Comment:
Can we read the projection from the s2 item collection?

Reply:
I assigned this to the same projection used for the S2 item.